File: sorting_algorithms/bubble_sort.py
# Implementation of Bubble Sort


def bubble_sort(L):
    n = len(L)

    for i in range(n):

        for j in range(n-i-1):
            if L[j] > L[j+1]:
                temp = L[j]
                L[j] = L[j+1]
                L[j+1] = temp

    return L



if __name__ == "__main__":
    L = [11, 21, 1, 7, 33, 2, 21, 99, 102]
    print(bubble_sort(L))

File: sorting_algorithms/bubble_sort.c
// Implementation of Bubble Sort

#include <stdio.h>

void bubble_sort(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n-i-1; j++)
        {
            if (arr[j] > arr[j+1])
            {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

void print_array(int arr[], int n)
{
    printf("[");
    for (int i = 0; i < n; i++)
    {
        printf("%d", arr[i]);
        if (i < n-1)
        {
            printf(",");

        }
    }
    printf("]");
}


int main()
{
    int arr[] = {101, 21, 44, 32, 47, 2};
    int n = sizeof(arr) / sizeof(arr[0]);

    printf("Before sorting: ");
    print_array(arr, n);

    bubble_sort(arr, n);

    printf("\nAfter bubble sorting: ");
    print_array(arr, n);
    printf("\n");
}